src/main/java/mekanism/common/CommonEventHandler.java

```diff
@@ -0,0 +1,32 @@
+package mekanism.common;
+
+import mekanism.common.item.ItemJetpack;
+import net.minecraftforge.fml.common.Mod.EventHandler;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+import net.minecraftforge.fml.relauncher.Side;
+
+/**
+ * @author ThatGamerBlue
+ */
+public class CommonEventHandler {
+
+    private boolean canFly = false;
+
+    @EventHandler
+    public void onPlayerTick(TickEvent.PlayerTickEvent event){
+        if(event.phase == TickEvent.Phase.START && event.side == Side.SERVER){
+            if(canFly){
```

Comment:
Seems like you cache a value on the serverside.
One for all players, while this event is called for each player.
I'm highly doubtful about this working properly.